application.go

```diff
@@ -122,10 +122,11 @@ func (r *Application) AllTaskRunning() bool {
 	if r.Tasks == nil {
 		return false
 	}
-	if r.TasksRunning == r.Instances {
-		return true
+	if r.HasHealthChecks() {
+		return r.TasksRunning == r.Instances && r.TasksHealthy == r.Instances
```

Comment:
The way I understand [Marathon health checks](https://mesosphere.github.io/marathon/docs/health-checks.html) is that every application has a health check, with the default one being Mesos' knowledge of the task state. In that sense, shouldn't we always have a properly set `r.TasksHealthy` value? If that's the case, then we should be able to just replace `r.TasksRunning` in the original if-expression by `r.TasksHealthy`, thus simplifying the entire if-clause to

```
if r.TasksHealthy == r.Instances {
  return true
}
```

I should be easy to get persuaded on this issue though as I have zero knowledge running Marathon application that do not come with a health check.